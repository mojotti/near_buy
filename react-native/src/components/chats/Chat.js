import React from 'react';
import { View } from 'react-native';


export default class Chat extends React.Component {
  static navigationOptions = ({ navigation }) => {
    return { title: navigation.state.params.item.title };
  };

  render() {
    console.log('thos', this.props);
    return (
      <View />
    );
  }
}
