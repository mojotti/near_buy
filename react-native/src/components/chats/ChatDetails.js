import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Dimensions, Text, TouchableOpacity, View } from 'react-native';
import { connect } from 'react-redux';
import { withNavigation } from 'react-navigation';
import { styles } from '../../static/styles/ListViewItemStyle';
import { localhost } from '../../static/constants';
import ImagePlaceholder from '../image_placeholder/ImagePlaceholder';

const WIDTH = Dimensions.get('window').width;

export class _ChatDetails extends PureComponent {
  _navigateToChat = () => {
    console.log('navigation', this.props)
    const { navigate } = this.props.navigation;
    navigate('Chat', { item: this.props.item.item });
  };

  render() {
    const id = this.props.item.item.item_id;
    const imagePath = `http://${localhost}:5000/api/v1.0/${id}/image0.jpg`;

    return (
      <View>
        <TouchableOpacity onPress={this._navigateToChat}>
          <View style={styles.itemContainer}>
            <View style={styles.imagePlaceholderContainer}>
              <View style={styles.imageContainer}>
                <ImagePlaceholder
                  styles={styles.image}
                  url={imagePath}
                  placeholderSize={WIDTH * 0.15}
                />
              </View>
            </View>
            <View style={styles.item}>
              <Text style={styles.title}>{this.props.item.item.title}</Text>
            </View>
          </View>
        </TouchableOpacity>
      </View>
    );
  }
}

_ChatDetails.propTypes = {
  item: PropTypes.shape({
    item: PropTypes.shape({
      item_id: PropTypes.number.isRequired,
      title: PropTypes.string.isRequired,
      seller_id: PropTypes.number.isRequired,
      buyer_id: PropTypes.number.isRequired,
    }).isRequired,
  }).isRequired,
};

const mapStateToProps = (state, ownProps) => {
  return {};
};

const ChatDetails = connect(mapStateToProps, null)(_ChatDetails);
export default withNavigation(ChatDetails);
