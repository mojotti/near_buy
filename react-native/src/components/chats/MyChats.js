import React from 'react';
import { FlatList, TouchableOpacity, View } from 'react-native';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Icon from 'react-native-vector-icons/MaterialIcons';

import MaterialIconAndText from '../common/MaterialIconAndText';
import {
  NO_CHATS_TEXT_HEADER,
  NO_CHATS_TEXT_CONTENT,
} from '../../static/constants';
import { requestChats } from '../../redux/actions/ChatActions';
import Chat from './ChatDetails';
import ItemSeparator from '../user_items/ItemSeparator';

export class _MyChats extends React.Component {
  static navigationOptions = ({ navigation }) => {
    return {
      title: 'Chats',
      headerLeft: (
        <View style={{ paddingHorizontal: 10 }}>
          <TouchableOpacity onPress={() => navigation.navigate('DrawerOpen')}>
            <Icon name="menu" size={30} color="#4d4dff" />
          </TouchableOpacity>
        </View>
      ),
    };
  };

  componentWillMount() {
    this._updateChats();
  }

  _updateChats = () => this.props.dispatch(requestChats(this.props.token));

  _renderNoChats = () => {
    return (
      <View style={{ flex: 1 }}>
        <MaterialIconAndText
          iconName="emoticon-poop"
          headerText={NO_CHATS_TEXT_HEADER}
          containerText={NO_CHATS_TEXT_CONTENT}
        />
      </View>
    );
  };

  _renderChat = rowData => <Chat item={rowData} itemId={rowData.index} />;

  _renderSeparator = () => <ItemSeparator widthPercentage={0.93} />;

  _renderChats = () => {
    return (
      <FlatList
        data={this.props.chatHeaders}
        renderItem={this._renderChat}
        keyExtractor={(item, index) => index.toString()}
        onRefresh={this._updateChats}
        refreshing={this.props.isFetching}
        ItemSeparatorComponent={this._renderSeparator}
      />
    );
  };

  render() {
    const isVisible =
      this.props.chatHeaders && this.props.chatHeaders.length > 0;
    const backgroundColor = isVisible ? '#FFFFFF' : 'transparent';
    const bgColor = { backgroundColor };

    return (
      <View style={[bgColor, { flex: 1 }]}>
        {isVisible ? this._renderChats() : this._renderNoChats()}
      </View>
    );
  }
}

_MyChats.propTypes = {
  isFetching: PropTypes.bool.isRequired,
  token: PropTypes.string.isRequired,
  chatHeaders: PropTypes.arrayOf(
    PropTypes.shape({
      seller_id: PropTypes.number.isRequired,
      buyer_id: PropTypes.number.isRequired,
      title: PropTypes.string.isRequired,
      item_id: PropTypes.number.isRequired,
    }).isRequired
  ).isRequired,
};

const mapStateToProps = state => {
  const { chatHeaders, isFetching } = state.currentChatsReducer;
  const { token } = state.authorizationReducer;

  return { chatHeaders, isFetching, token };
};

const MyChats = connect(mapStateToProps)(_MyChats);
export default MyChats;
