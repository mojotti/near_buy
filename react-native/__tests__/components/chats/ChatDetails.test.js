import React from 'react';
import { shallow } from 'enzyme';
import { _ChatDetails } from '../../../src/components/chats/ChatDetails';

const item = {
  item: {
    id: 1,
    title: 'foo',
    seller_id: 0,
    buyer_id: 0,
  }
};

describe('<ChatDetails />', () => {
  test('renders correctly', () => {
    const wrapper = shallow(<_ChatDetails item={item} />);
    expect(wrapper).toMatchSnapshot();
  });
});
