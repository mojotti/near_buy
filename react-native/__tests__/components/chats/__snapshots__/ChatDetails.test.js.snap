// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatDetails /> renders correctly 1`] = `
<Component>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
  >
    <Component
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderWidth": 0,
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <Component
        style={
          Object {
            "flexDirection": "column",
            "height": 150,
            "justifyContent": "flex-start",
            "width": 150,
          }
        }
      >
        <Component
          style={
            Object {
              "alignSelf": "center",
              "borderRadius": 100,
              "height": 112.5,
              "justifyContent": "center",
              "margin": 10,
              "overflow": "hidden",
              "width": 112.5,
            }
          }
        >
          <ImagePlaceholder
            placeholderSize={112.5}
            styles={
              Object {
                "height": 112.5,
                "width": 112.5,
              }
            }
            url="http://localhost:5000/api/v1.0/1/image0.jpg"
          />
        </Component>
      </Component>
      <Component
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "flex": 1,
            "flexDirection": "column",
            "padding": 10,
            "width": 750,
          }
        }
      >
        <Component
          style={
            Object {
              "fontSize": 17,
              "fontWeight": "bold",
            }
          }
        >
          foo
        </Component>
      </Component>
    </Component>
  </TouchableOpacity>
</Component>
`;
